### PR TITLE
DBTP-617 - Add command to trigger pipeline based build

### PR DESCRIPTION
**Pull request by yusufsheiqh**
## Context

- Add `codebase build` command to trigger pipeline based build

### CODE REVIEW

**Review by acodeninja — APPROVED** [review]
Small suggestion, otherwise great

File: dbt_copilot_helper/commands/codebase.py [hunk 1]
```diff
@@ -74,3 +79,45 @@ def prepare():
         phase_contents = templates.get_template(f".copilot/phases/{phase}.sh").render()
 
         click.echo(mkfile(Path("./.copilot"), f"phases/{phase}.sh", phase_contents, overwrite=True))
+
+
+@codebase.command(cls=ClickDocOptCommand)
```
[the remark this answers is not in the document]
Comment (yusufsheiqh) [reply dbt_copilot_helper/commands/codebase.py]: Aaand it's gone! Resolved in https://github.com/uktrade/copilot-tools/pull/255/commits/f8557cd3a8e7e25ecf6dec6a453a34e8878541c4.

File: dbt_copilot_helper/commands/codebase.py [hunk 2]
```diff
@@ -74,3 +78,45 @@ def prepare():
...
+        )
+        raise click.Abort
+
+    check_if_commit_exists = subprocess.run(
+        ["git", "branch", "-r", "--contains", f"{commit}"], capture_output=True, text=True
+    )
+
+    if check_if_commit_exists.stderr:
+        click.secho(
+            f"""The commit hash "{commit}" either does not exist or you need to run `git pull`.""",
```
Comment (antroy-madetech) [inline dbt_copilot_helper/commands/codebase.py:104]: This can just be `git fetch`

File: dbt_copilot_helper/commands/codebase.py [hunk 3]
```diff
@@ -74,3 +78,45 @@ def prepare():
...
+        f"""You are about to build "{app}" for "{codebase}" with commit "{commit}". Do you want to continue?"""
+    ):
+        boto3.client("codebuild").start_build(
+            projectName=f"codebuild-{app}-{codebase}",
+            artifactsOverride={"type": "NO_ARTIFACTS"},
+            sourceVersion=commit,
+        )
+
+        return click.echo(
+            "Your build has been triggered and you can check your build progress in the AWS Console."
```
Comment (antroy-madetech) [inline dbt_copilot_helper/commands/codebase.py:119]: May read better as: "Your build has been triggered. Check your build progress in the AWS Console."

**Review by antroy-madetech — APPROVED** [review]

File: dbt_copilot_helper/commands/codebase.py (hunk 3 above)
Comment (acodeninja) [reply dbt_copilot_helper/commands/codebase.py:119]: Another option could be to link to the build, though the user would also need to be prompted to log into the AWS account. See the `get_build_url` function in [ci-image-builder](https://github.com/uktrade/ci-image-builder/blob/main/image_builder/notify.py#L118)

File: dbt_copilot_helper/commands/codebase.py (hunk 2 above)
Comment (yusufsheiqh) [reply dbt_copilot_helper/commands/codebase.py]: This makes sense to me, so we don't override peeps' local repos. Resolved in https://github.com/uktrade/copilot-tools/pull/255/commits/28d645a856e80cb8ea9ec1c977a4561062fda34b.

File: dbt_copilot_helper/commands/codebase.py (hunk 3 above)
Comment (yusufsheiqh) [reply dbt_copilot_helper/commands/codebase.py]: Went with Ant's suggested change for now! Resolved in https://github.com/uktrade/copilot-tools/pull/255/commits/28d645a856e80cb8ea9ec1c977a4561062fda34b.

**Pull request merged**